confection/__init__.py

```diff
@@ -912,6 +914,15 @@ def _fill(
                     # created via config blocks), only use its values
                     validation[v_key] = list(validation[v_key].values())
                     final[key] = list(final[key].values())
+
+                    if ARGS_FIELD_ALIAS in schema.__fields__ and not resolve:
+                        # If we're not resolving the config, make sure that the field
+                        # expecting the promise is typed Any so it doesn't fail
+                        # validation if it doesn't receive the function return value
+                        field = schema.__fields__[ARGS_FIELD_ALIAS]
+                        schema.__fields__[ARGS_FIELD_ALIAS] = copy_model_field(
+                            field, Any
+                        )
```

Comment:
This is a copy of a logic appearing earlier (if it is a promise). It seems like it might be possible to move it outside the 'if promise'-scope.